lib/commands/recordscreen.js

```diff
@@ -0,0 +1,144 @@
+*                if it is -1, the default value is 4Mbps. You can increase the bit rate to improve video quality,
+*                but doing so results in larger movie files.
+*                for example, 6000000
+*
+*/
+commands.startRecordingScreen = async function ( filePath, videoSize, timeLimit, bitRate ) {
+  let cmd, data, length, index, indexNoSuchFile;
+  let loop = true;
+
+  cmd = ['getprop', 'ro.build.version.sdk'];
+  data = await adb.shell(cmd); 
+
+  //this function is suppported on the device running android 4.4(api level 19)
+  if (data < 19) throw new Error('The screenrecord command is able to record the display of devices running Android 4.4 (API level 19) and higher.');
+
+  cmd = ['ls', '\'' + filePath + '\'', '|', 'grep', '\'No such file\''];
+  data = await adb.shell(cmd); 
+
+  length = _.size(data);
+
+  //if there's same file in the path, then thorws error
+  if (length === 0) throw new Error('\'' + filePath + '\' is already exist.');
+
+
+  //make adb command 
+  cmd = ['screenrecord', '\'' + filePath + '\''];
+
+  index = 2;
+
+  if ( videoSize !== "default" ) {
+    cmd [index ++] = "--size";
+    cmd [index ++] = videoSize;
+  }  
+
+  if ( timeLimit !== -1 ) {
+    cmd [index ++] = "--time-limit";
+    cmd [index ++] = timeLimit;
+  } 
+
+  if ( bitRate !== -1 ) {
+    cmd [index ++] = "--bit-rate";
+    cmd [index ++] = bitRate;
+  } 
+
+  //call the command  
+  data = adb.shell(cmd); 
+
+  
+  //there is the delay time to start recording the screen, so, wait until it is ready.
+  //the ready condition is
+  //1. check the movie file is created
+  //2. check it is started to capture the screen 
+  cmd = ['ls', '-al', filePath]; 
+  while (loop) {
+    data = await adb.shell(cmd); 
+    indexNoSuchFile = data.indexOf("No such file");
+
+    if (indexNoSuchFile < 0) {
+      var arrayList2 = data.split(" ");
+      length = _.size(arrayList2);
+      var availableDataIndex = 0;
+      
+      for (let i = 0 ; i < length ; ++ i){
+        if ( _.size(arrayList2[i]) > 0 ){
+          availableDataIndex++;
+          //check it is started to capture the screen
+          if (availableDataIndex === 4){
+            if ( (arrayList2[i] * 1) > 32 )
+              return;
+          }
+        }
+      }
+      await sleep(1000);
+    } else
+      await sleep(1000);
+  } 
+};
+
+/**
+* stop recording the screen.
+*/
+commands.stopRecordingScreen = async function ( ) {
+  let cmd, data, length;
+  let loop = true;
+
+  //check the pid of the screenrecord process 
+  cmd = ['ps', '|', 'grep', 'screenrecord'];
+  data = await adb.shell(cmd); 
+
+  length = _.size(data);
+
+  if (length > 0){
+    var arrayList2 = data.split(" ");
+    length = _.size(arrayList2);
+
+    //kill the process and wait for that it is completed
+    for (let i = 1 ; i < length ; ++ i){
+      if ( _.size(arrayList2[i]) > 0 ){
+        var screenrecordPID = arrayList2[i] * 1;
+
+        while (loop) {
```

Comment:
I tested your code,
the function works asynchronously, so it returns to client at the first try.
so, I have to change the performance code which is previous pull request.